### PR TITLE
rbd: help message distinction between commands and aliases

### PR DESCRIPTION
**Pull request by HeyoQiang**
Fixes BUG #15521:Seeing rollback and revert option for snapshot, which seems to be same and can be confusing for users

Signed-off-by: Yongqiang He <he.yongqiang@h3c.com>

### CODE REVIEW

**Comment by dillaman** [comment]
Perhaps as an alternative suggestion, the per-command help page could include the alias (if present):

> $ rbd help snap revert
> usage: rbd snap rollback [--pool <pool>] [--image <image>] [--snap <snap>] 
>                          [--no-progress] 
>                          <snap-spec> 
> alias: rbd snap revert
> 
> Rollback image to snapshot.
> 
...

**Comment by dillaman** [comment]
You most likely will have to update the `src/test/cli/rbd/help.t` test output -- try running the ```test/run-cli-tests``` test

**Comment by HeyoQiang** [comment]
@dillaman I think we can keep it in original format.

File: src/tools/rbd/Shell.cc [hunk 1]
```diff
@@ -302,14 +305,19 @@ void Shell::print_help() {
...
+ }
 
+void Shell::print_action_help(Action *action, bool is_alias) {
   std::stringstream ss;
-  ss << "usage: " << APP_NAME << " "
-     << format_command_spec(action->command_spec);
-  std::cout << ss.str();
+  if (is_alias == false) {
+    ss << "usage: " << APP_NAME << " "
+       << format_command_spec(action->command_spec);
```
Comment (dillaman) [inline src/tools/rbd/Shell.cc]: Probably a good use of the ternary operator to eliminate the dup code: ```format_command_spec(is_alias ? action->alias_command_spec : action->command_spec)```

File: src/tools/rbd/Shell.h [hunk 2]
```diff
@@ -57,12 +57,12 @@ class Shell {
   void get_command_spec(const std::vector<std::string> &arguments,
                         std::vector<std::string> *command_spec);
   Action *find_action(const CommandSpec &command_spec,
-                      CommandSpec **matching_spec);
+                      CommandSpec **matching_spec, bool &is_alias);
```
Comment (dillaman) [inline src/tools/rbd/Shell.h]: Nit: should be a pointer per the coding style

File: src/tools/rbd/Shell.cc (hunk 1 above)
Comment (HeyoQiang) [inline src/tools/rbd/Shell.cc]: @dillaman  lgtm.

File: src/tools/rbd/Shell.h (hunk 2 above)
Comment (HeyoQiang) [inline src/tools/rbd/Shell.h]: @dillaman I have changed it

**Comment by dillaman** [comment]
@HeyoQiang The latest commit doesn't compile.

**Comment by dillaman** [comment]
@HeyoQiang Last comment: can you change your commit message to include:

> Fixes: https://tracker.ceph.com/issues/15521

instead of:

> Fix BUG #15521 :Seeing rollback and revert option for snapshot, which seems to be same and can be confusing for users

**Comment by HeyoQiang** [comment]
@dillaman I have changed it.

**Pull request merged**